linkerd/tls/src/main/scala/io/buoyant/linkerd/tls/TlsUtils.scala

```diff
@@ -22,11 +24,22 @@ object TlsUtils {
 
   }
   case class Certs(caCert: File, serviceCerts: Map[String, ServiceCert])
+
   def withCerts(names: String*)(f: Certs => Unit): Unit = {
+    withCerts(false, names)(f: Certs => Unit)
```

Comment:
`true` should be the default value. If I understand your comment below, some tests will fail if `true` is used. I think that instead we should modify *those* tests to explicitly pass in `false`.

In particular, we should move away from using the subject CN for anything except as a fallback when the DNS names aren't provided, following RFC 6125.

Reply:
Sounds good, will make changes accordingly.